From 572862e0f8a6611d4a7d4b2f4fb4da4339a84961 Mon Sep 17 00:00:00 2001
From: Julien Thomas <jthomas@zenetys.com>
Date: Fri, 21 Jun 2024 20:27:49 +0200
Subject: [PATCH 1/2] fix(cifs): Pass options port and timeout to SmbClient
 constructor

Centreon patched version of Filesys::SmbClient perl module supports
custom port and timeout options and those are advertised in usage help
but ignored, let's fix it.
---
 src/apps/protocols/cifs/custom/libcifs.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/apps/protocols/cifs/custom/libcifs.pm b/src/apps/protocols/cifs/custom/libcifs.pm
index 9c14393de..b8e2f8b81 100644
--- a/src/apps/protocols/cifs/custom/libcifs.pm
+++ b/src/apps/protocols/cifs/custom/libcifs.pm
@@ -69,7 +69,7 @@ sub check_options {
 
     $self->{hostname} = (defined($self->{option_results}->{hostname})) ? $self->{option_results}->{hostname} : '';
     $self->{port} = (defined($self->{option_results}->{port})) ? $self->{option_results}->{port} : 139;
-    $self->{timeout} = (defined($self->{option_results}->{timeout})) && $self->{option_results}->{timeout} =~ /^\d+$/? $self->{option_results}->{timeout} : 30;
+    $self->{timeout} = (defined($self->{option_results}->{timeout})) && $self->{option_results}->{timeout} =~ /^\d+$/? $self->{option_results}->{timeout} * 1000 : 30000;
     $self->{cifs_username} = (defined($self->{option_results}->{cifs_username})) ? $self->{option_results}->{cifs_username} : '';
     $self->{cifs_password} = (defined($self->{option_results}->{cifs_password})) ? $self->{option_results}->{cifs_password} : '';
     $self->{workgroup} = (defined($self->{option_results}->{workgroup})) ? $self->{option_results}->{workgroup} : '';
@@ -90,6 +90,8 @@ sub init_cifs {
             username  => $self->{cifs_username},
             password  => $self->{cifs_password},
             workgroup => $self->{workgroup},
+            port => $self->{port},
+            timeout => $self->{timeout},
             debug => $self->{output}->is_debug() ? 10 : 0
         );
     }
-- 
2.21.1

